app/src/main/java/org/oppia/app/home/HomeFragmentPresenter.kt

```diff
@@ -57,6 +68,10 @@ class HomeFragmentPresenter @Inject constructor(
     itemList.add(allTopicsViewModel)
     topicListAdapter = TopicListAdapter(activity, itemList, promotedStoryList)
 
+    id = activity.intent.getIntExtra(KEY_HOME_PROFILE_ID, 0)
```

Comment:
Why are we still passing id via intent, we should be using `profileManagementController.getCurrentProfileId()`

Reply:
With intent it returns correct id, where as if we use profileManagementController.getCurrentProfileId() it is giving -1 in testcases. To fix that we are using id through intent.